packages/core/src/browser/i18n/language-quick-pick-service.ts

```diff
@@ -0,0 +1,124 @@
+// *****************************************************************************
+// Copyright (C) 2022 TypeFox and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { inject, injectable } from 'inversify';
+import { nls } from '../../common/nls';
+import { AsyncLocalizationProvider, LanguageInfo } from '../../common/i18n/localization';
+import { QuickInputService, QuickPickItem, QuickPickSeparator } from '../quick-input';
+import { WindowService } from '../window/window-service';
+
+export interface LanguageQuickPickItem extends QuickPickItem {
+    languageId: string
+    execute?(): Promise<void>
+}
+
+@injectable()
+export class LanguageQuickPickService {
+
+    @inject(QuickInputService) protected readonly quickInputService: QuickInputService;
+    @inject(AsyncLocalizationProvider) protected readonly localizationProvider: AsyncLocalizationProvider;
+    @inject(WindowService) protected readonly windowService: WindowService;
+
+    async pickDisplayLanguage(): Promise<string | undefined> {
+        const quickInput = this.quickInputService.createQuickPick();
+        const installedItems = await this.getInstalledLanguages();
+        const quickInputItems: (QuickPickItem | QuickPickSeparator)[] = [
+            {
+                type: 'separator',
+                label: nls.localize('theia/core/installedLanguages', 'Installed languages')
+            },
+            ...installedItems
+        ];
+        quickInput.items = quickInputItems;
+        quickInput.busy = true;
```

Comment:
Do we have any visual indicator that reflects this state? When I tested this, the experience was a bit jarring: a moment or two of just the installed languages with the available languages suddenly popping up - I couldn't see any indication that anything was happening in the background. VSCode isn't much better, but it would be nice to have something.

Reply:
It seems like the busy indicator for quick inputs in VSCode has been broken for two years (starting in v1.44), see https://github.com/microsoft/vscode/issues/128966.

However, the actual html element is correctly added to the DOM, so with some CSS magic we're able to correctly display the busy indicator :)